File: KantaroWpf/Converters/InvertBoolConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Enzo.Music.KantaroWpf.Converters;


[ValueConversion(typeof(bool), typeof(bool))]
public partial class InvertBoolConverter : IValueConverter
{

    public InvertBoolConverter()
    {

    }

    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        if (value is not null && value is bool)
        {
            return !(bool)value;
        }

        return true;
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        return Convert(value, targetType, parameter, culture);
    }
}

File: CanzoniereWPF/Converters/InvertBoolConverter.vb
Imports System.Globalization

<ValueConversion(GetType(Boolean), GetType(Boolean))>
Public Class InvertBoolConverter
    Implements IValueConverter

    Public Sub New()

    End Sub

    Private Function IValueConverter_Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If value IsNot Nothing AndAlso TypeOf value Is Boolean Then
            Return Not CBool(value)
        End If

        Return True
    End Function

    Private Function IValueConverter_ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return IValueConverter_Convert(value, targetType, parameter, culture)
    End Function
End Class